tests/unit_tests/test_runner.py

```
@@ -67,6 +67,16 @@ def test_initialize_model(tmp_path, mgf_small):
         runner.initialize_model(train=False)
 
 
+def test_db_search_no_model_raises(tiny_config_db, mgf_small, tiny_fasta_file):
+    """DB search must require an explicit model file."""
+    config = Config(tiny_config_db)
+    with ModelRunner(config=config) as runner:
+        with pytest.raises(
+            ValueError, match="A model file must be provided for DB search"
+        ):
```

Ruff (0.15.6)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)